src/main/java/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -192,32 +192,21 @@ public static int getTransportCost(final Collection<Unit> units) {
   }
 
   private static List<Unit> sortByTransportCapacityDescendingThenMovesDescending(final Collection<Unit> transports) {
-    final Comparator<Unit> transportCapacityComparator = (o1, o2) -> {
-      final int capacityLeft1 = TransportTracker.getAvailableCapacity(o1);
-      final int capacityLeft2 = TransportTracker.getAvailableCapacity(o2);
-      if (capacityLeft1 != capacityLeft2) {
-        return Integer.compare(capacityLeft2, capacityLeft1);
-      }
-      final int movementLeft1 = TripleAUnit.get(o1).getMovementLeft();
-      final int movementLeft2 = TripleAUnit.get(o2).getMovementLeft();
-      return Integer.compare(movementLeft2, movementLeft1);
-    };
     final List<Unit> canTransport = new ArrayList<>(transports);
-    Collections.sort(canTransport, transportCapacityComparator);
+    Collections.sort(canTransport, Comparator.comparingInt(TransportTracker::getAvailableCapacity)
+        .thenComparing(TripleAUnit::get,
+            Comparator.comparingInt(TripleAUnit::getMovementLeft)));
```

Comment:
It looks like the values passed to the two `Integer#compare()` calls above are reversed.  That is, the value for `o2` is passed as the first parameter, and the value for `o1` is passed as the second parameter.  Should the resulting objects returned by `Comparator#comparingInt()` be reversed?  (Note that the method name specifies "descending" on both sorts.)

Reply:
Good catch. I missed that.